awx/main/models/ha.py

```diff
@@ -208,6 +208,15 @@ def jobs_running(self):
     def jobs_total(self):
         return UnifiedJob.objects.filter(instance_group=self).count()
 
+    @property
+    def is_controller(self):
+        return self.controlled_groups.exists()
+
+    @property
+    def is_isolated(self):
+        return all([instance.is_isolated for instance in self.instances.all()]) \
```

Comment:
This looks like it should work. Checking all of the instances gave me pause though because TMK we don't allow mixed groups (that have both isolated and regular / full instances), so it seems like you could just do a check on the first instance you come across. I don't know that there's much of a penalty for querying all of the instances, though, so not sure that there's a big reason to change what's here. Just something that came to mind.

Reply:
This is making an unnecessary traversal. This is the InstanceGroup model, so the logic for this should be simply

```python
    @property
    def is_isolated(self):
        return bool(self.controller)
```

This information is already available to clients, so it's not necessary to provide this field for them to have all the information, but it's self-obvious to have a field named this.

Reply:
this is still using unneeded complicated logic which may not be correct in all cases.

Reply:
@AlanCoding marked as resolved a bit early, my bad